src/composables/widgets/store.ts

```diff
@@ -12,14 +12,14 @@ Contributors: Smart City Jena
 import { ref, type Ref, inject } from "vue";
 import type { TinyEmitter } from "tiny-emitter";
 
-export function useStore<Type extends IStore>(updateFn?, watcher?, externalEventBus?) {
+export function useStore<Type extends IStore>(
+    eventBus: TinyEmitter,
```

Comment:
this breaks several widget implementations using the old signature --> go back optional on 3 place

Reply:
In that case I need to rewrite old widgets implementations. EventBus in core part of the store. It can't be injected from composable, so the only way to set it is to pass as parameter.

Reply:
if u want to make eventbus is mandatory all old widgets hat to be rewriten bause they use useStore() without parameters  e.g RepaetableSVGWiget: L51: const { store, data, setStore } = useStore<Store>();